test/setup/cluster_setup.go

```diff
@@ -386,45 +387,59 @@ func (c *Cluster) createUsersCryptoMaterials() error {
 	}
 
 	for _, user := range c.Users {
-		pemUserCert, pemUserKey, err := testutils.IssueCertificate("Cluster User: "+user, "127.0.0.1", keyPair)
+		c.CreateOrUpdateUserCerts(user, keyPair)
+	}
 
-		userCertPath := path.Join(c.testDirAbsPath, "users", user+".pem")
-		pemUserCertFile, err := os.Create(userCertPath)
-		if err != nil {
-			return err
-		}
-		_, err = pemUserCertFile.Write(pemUserCert)
-		if err != nil {
-			return err
-		}
-		if err = pemUserCertFile.Close(); err != nil {
-			return err
-		}
+	return nil
+}
 
-		userKeyPath := path.Join(c.testDirAbsPath, "users", user+".key")
-		pemUserKeyFile, err := os.Create(userKeyPath)
+func (c *Cluster) CreateOrUpdateUserCerts(user string, keyPair tls.Certificate) error {
+	var err error
+	if reflect.ValueOf(keyPair).IsZero() {
+		keyPair, err = tls.X509KeyPair(c.rootCAPemCert, c.caPrivKey)
```

Comment:
This pattern is confusing... in other packages it seems it is always "Update", whereas internally in the test package it is always create. Can we provide two methods? they may map to an internal "ugly" method, but passing in an empty key as flag for Update is a bit obscure for my taste.

Reply:
Done